openfl/utilities/verifdataset/verifiable_dataset.py

```diff
@@ -0,0 +1,164 @@
+# Copyright 2020-2025 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+""" This module contains the VerifiableDataset class. """
+
+import json
+from enum import Enum
+from hashlib import sha384
+from pathlib import Path
+from typing import List
+
+from openfl.utilities.verifdataset.data_source import DataSource, DataSourceType
+from openfl.utilities.verifdataset.local_data_source import LocalDataSource
+
+
+class DatasetFormat(Enum):
+    """Enum for the different dataset commitment formats."""
+
+    VERBOSE = "verbose_dataset"
+    CONCISE = "concise_dataset"
```

Comment:
Following the docstring, I suggest renaming to `DatasetCommitment`. Down the road, we could imagine adding a MERKLE_TREE entry here.